### PR TITLE
Canvas UI tests for inbox

### PR DESCRIPTION
**Pull request by GabrielPalmer**
refs: MBL-12589, MBL-12488, MBL-12587, MBL-12586
affects: student
release note: none

### CODE REVIEW

File: Canvas/CanvasUITests/Inbox/InboxTests.swift [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    func testCanFilterMessagesAndShowsUnread() {
+        //Dashboard
+        XCTAssert(Dashboard.inboxTab.value == "2 items")
+        Dashboard.inboxTab.tap()
+
+        // Inbox
+        Inbox.filterButton.tap()
+        Inbox.filterOption("Assignment").waitToExist()
+        Inbox.filterOption("Assignment").tap()
+        Inbox.message(id: "47").waitToVanish()
```
Comment (armstrongnate) [inline Canvas/CanvasUITests/Inbox/InboxTests.swift]: Make sure that this message exists before we test that it vanishes.

**Comment by armstrongnate** [comment]
To fix the Danger build you will need to run `yarn test -u` from `rn/Teacher/` to update the test snapshots.

Looks like CanvasUITests failed on `testProfileChangesUser()` which is probably not related to these changes.

File: Canvas/CanvasUITests/Inbox/InboxTests.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    static func messageAllStudents(courseID: String) -> Element {
+        return app.find(id: "branch_course_\(courseID)_students")
+    }
+
+    static func student(studentID: String) -> Element {
+        return app.find(id: studentID)
+    }
+}
```
Comment (vanwagonet) [inline Canvas/CanvasUITests/Inbox/InboxTests.swift]: All the above would be better in another file `Inbox.swift`.

Comment (GabrielPalmer) [reply Canvas/CanvasUITests/Inbox/InboxTests.swift]: Got that done. Any idea on the issues with yarn?

Comment (armstrongnate) [reply Canvas/CanvasUITests/Inbox/InboxTests.swift]: It is weird that the snapshots are off. Maybe run `yarn test -u` again (from rn/Teacher) and see if the snaps update _again_? Could be a difference in node maybe. I'm not really sure.

Comment (GabrielPalmer) [reply Canvas/CanvasUITests/Inbox/InboxTests.swift]: Already tried that. Doesn't update anything this time. I merged master into my branch next and pushed it up but that didn't make a difference either.

**Comment by armstrongnate** [comment]
@GabrielPalmer `yarn test` fails for me locally on this branch (same snapshot error as bitrise). Now we need to figure out why it passes on your machine...

From `rn/Teacher/`, first run `yarn install` to make sure you're on the correct version of all the dependencies then run `yarn test` again.

Then tell me the output of these:

```
node -v
yarn -v
```

**Comment by GabrielPalmer** [comment]
node -v
v12.4.0

yarn -v
1.16.0

**Comment by armstrongnate** [comment]
Me:
* Node.js: v8.9.0
* Yarn: 1.3.2

Bitrise:
* Node.js: v10.15.2
* Yarn: 1.15.2

Apparently we are all on different versions. My best guess right now is that it's the node version. I think we probably need to look into why the quizzes are suddenly getting re-ordered.

**Comment by vanwagonet** [comment]
Yeah, the node version is the most likely culprit. I'm running `v10.15.3`

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Canvas/CanvasUITests/Assignments/Assignments.swift
* Canvas/CanvasUITests/Inbox/Inbox.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12589](https://instructure.atlassian.net/browse/MBL-12589)
[MBL-12488](https://instructure.atlassian.net/browse/MBL-12488)
[MBL-12587](https://instructure.atlassian.net/browse/MBL-12587)
[MBL-12586](https://instructure.atlassian.net/browse/MBL-12586)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.16% | 95.16% | 0%
**Student** | 92.69% | 92.69% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.78% | 92.78% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**